theano/tensor/signal/downsample.py

```diff
@@ -518,6 +600,15 @@ def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
                 " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
         self.mode = mode
 
+    def infer_shape(self, node, in_shapes):
+        return [in_shapes[0]]
+
+
+class MaxPoolGrad(PoolGrad):
```

Comment:
We pass a mode that force that optimization to run. Pass this parameter to
theano.function

mode=Theano.compile.get_default_mode().including('optname')
Le 5 août 2015 20:11, "sebastien-j" <notifications@github.com> a écrit :

> In theano/tensor/signal/downsample.py
> <https://github.com/Theano/Theano/pull/3139#discussion_r36370048>:
>
> > @@ -518,6 +600,15 @@ def __init__(self, ds, ignore_border, st=None, padding=(0, 0), mode='max'):
> >                  " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
> >          self.mode = mode
> >
> > +    def infer_shape(self, node, in_shapes):
> > +        return [in_shapes[0]]
> > +
> > +
> > +class MaxPoolGrad(PoolGrad):
>
> @nouiz <https://github.com/nouiz> , I added the optimization. However,
> tests fail with mode=FAST_COMPILE, where the optimization isn't applied. I
> suppose this is normal behaviour. Should I just skip the test in that case,
> or is there something better to do?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3139/files#r36370048>.
>